crypto/x509/v3_utl.c

```diff
@@ -1096,13 +1096,17 @@ int ossl_a2i_ipadd(unsigned char *ipout, const char *ipasc)
 
 static int ipv4_from_asc(unsigned char *v4, const char *in)
 {
-    int a0, a1, a2, a3;
+    const char *p;
+    int a0, a1, a2, a3, n;
 
-    if (sscanf(in, "%d.%d.%d.%d", &a0, &a1, &a2, &a3) != 4)
+    if (sscanf(in, "%d.%d.%d.%d%n", &a0, &a1, &a2, &a3, &n) != 4)
         return 0;
     if ((a0 < 0) || (a0 > 255) || (a1 < 0) || (a1 > 255)
```

Comment:
This condition is ugly....
On the other hand `(a0 | a1 | a2 | a3) & ~0377` is a thing of beauty.

No need to change.  Unless you agree :)

Reply:
LGTM, changed.

Reply:
Oh dear no, that is a horrible ugly hack that nobody will understand the first time they look at the code.  As you had it before, it was clean and understandable.  Please don't.

Reply:
Maybe adding a comment `/* all numbers must be within 0 and 255 */` would make it OK even for you @richsalz?

Reply:
@amiremohamadi could you please add that comment?

Reply:
I'm not on the project. I don't get why you're trying to save a couple of clock cycles.  Shrug.

Reply:
+1 to @richsalz's comment. Write the obvious thing. It won't matter and, besides, the compiler can figure this out for you:
https://godbolt.org/z/r4n1rr6Tf

Reply:
Would are more restrictive format like ``%hhu`` work or is it not portable / undefined for leading minus sign?

```
    unsigned char a0, a1, a2, a3;
    int n;

    if (sscanf(in, "%hhu.%hhu.%hhu.%hhu%n", &a0, &a1, &a2, &a3, &n) != 4)
```